src/coq_elpi_builtins.ml

```diff
@@ -941,6 +936,23 @@ let mkCLocalAssum x y z = Constrexpr.CLocalAssum(x,None,y,z)
 let pattern_of_glob_constr env g = Patternops.pattern_of_glob_constr env g
 [%%endif]
 
+[%%if coq = "8.19"]
+let declare_definition hack _using ~cinfo ~info ~opaque ~body sigma =
+  let gr = Declare.declare_definition ~cinfo ~info ~opaque ~body sigma in
+  gr, Option.get !hack
+[%%elif coq = "8.20"]
+let declare_definition hack using ~cinfo ~info ~opaque ~body sigma =
+  let using = Option.map Proof_using.using_from_string using in
+  let gr = Declare.declare_definition ~cinfo ~info ~opaque ~body ?using sigma in
+  gr, Option.get !hack
+[%%else]
+let declare_definition _ using ~cinfo ~info ~opaque ~body sigma =
+  let using = Option.map Proof_using.using_from_string using in
+  let gr, uctx = Declare.declare_definition_full ~cinfo ~info ~opaque ~body ?using sigma in
+  gr, UState.context_set uctx
```

Comment:
maybe need to do `if poly then empty else UState.context_set uctx` (not sure where `poly` is, I think either `cinfo` or `info`)